src/TConsole.cpp

```diff
@@ -2128,9 +2217,35 @@ void TConsole::setCaretMode(bool enabled)
             QAccessibleEvent event(mpCommandLine, QAccessible::Focus);
             QAccessible::updateAccessibility(&event);
         } else if (mType == UserWindow) {
-            mUpperPane->setFocusProxy(mpHost->mpConsole->mpCommandLine);
-            QAccessibleEvent event(mpHost->mpConsole->mpCommandLine, QAccessible::Focus);
-            QAccessible::updateAccessibility(&event);
+            if (mpCommandLine && mpCommandLine->isVisible()) {
+                mUpperPane->setFocusProxy(mpCommandLine);
```

Comment:
This PR is already going above & beyond the original ticket (that is great!) but this focus switching seems to be unrelated to the scope. Mind breaking this out into a separate PR for easier testing and better Git history?

Reply:
I'll try to extract it and move it to the ***other*** PR.

Reply:
Ah, I'm not sure I can do that - the various calls to `QAccessible::updateAccessibility(...)` are all about the accessibility system **announcing where the focus has gone to** which ***is*** tied into the subject of this PR...